flang/test/Lower/pointer-runtime.f90

```diff
@@ -0,0 +1,50 @@
+! RUN: bbc -emit-fir -use-alloc-runtime %s -o - | FileCheck %s
+
+! Test lowering of allocatables using runtime for allocate/deallocate statements.
+! CHECK-LABEL: _QPpointer_runtime(
+subroutine pointer_runtime(n)
+  integer :: n
+  character(:), pointer :: scalar, array(:)
```

Comment:
Allocation of a pointer to a character or even a pointer to an array of characters seems straightforward. Why do we need calls to a runtime function?

Reply:
For character, the default is to do the allocation with inlined code. This test file is testing the `-use-alloc-runtime` that forces the use of the runtime for all allocations.

By default, the allocation runtime is used for derived type because of the default initialization constraints related to derived type allocation that the runtime now deals with.